### ISSUE TITLE
Simulator should provide warnings or errors when CSP violations are detected in packaged apps

### ISSUE THREAD

**Issue by jds2501** [mozilla/r2d2b2g#684]
See the situation that happened on https://bugzilla.mozilla.org/show_bug.cgi?id=889356.

We had an app developer who built an app that produced odd behavior in the simulator that did not follow the CSP, but worked correctly on device with the CSP enforced. I'm filing a separate issue on that. However, this issue looks into providing warnings/errors to the user if the packaged app provided by the user violates the CSP for packaged apps.

**Comment by mykmelez** [comment]
Yup, it would be useful for the Simulator to detect that an app is going to violate the CSP and warn the user about that. Unfortunately, I'm not sure how actually to do that, since a reliable check would probably require parsing the HTML source of the app. However, we might be able to satisfy the 80% case by simply looking for common strings suggesting a violation (like on\* handler attributes).

**Comment by ochameau** [comment]
I think @jds2501 is just asking for runtime error to show up somehow, somewhere. The CSP component most likely throw some message in adb logcat, and if we are lucky, also in web console.
If these errors already appear in the web console, this request would be just about fixing issue #685.
Otherwise, I think someone, maybe from telefonica is working on a static parser for CSP.
